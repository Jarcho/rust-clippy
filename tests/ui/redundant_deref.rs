#![warn(clippy::redundant_deref)]

fn foo(_: &str) {}

fn main() {
    foo(&*String::new());
}
