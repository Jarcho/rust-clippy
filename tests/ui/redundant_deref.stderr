error: redundant dereference
  --> $DIR/redundant_deref.rs:6:9
   |
LL |     foo(&*String::new());
   |         ^^^^^^^^^^^^^^^ help: remove the dereference: `&String::new()`
   |
   = note: `-D clippy::redundant-deref` implied by `-D warnings`

error: aborting due to previous error

